internal/framework/resource_kubectl_manifest_deferred_test.go

```
@@ -0,0 +1,78 @@
+
+	modifyPlan := func(deferralAllowed bool) *resource.ModifyPlanResponse {
+		req := resource.ModifyPlanRequest{
+			Config:             tfsdk.Config{Schema: schemaResp.Schema, Raw: unknownCfg},
+			Plan:               tfsdk.Plan{Schema: schemaResp.Schema, Raw: planVal},
+			State:              tfsdk.State{Schema: schemaResp.Schema, Raw: nullState},
+			ClientCapabilities: resource.ModifyPlanClientCapabilities{DeferralAllowed: deferralAllowed},
+		}
+		resp := &resource.ModifyPlanResponse{
+			Plan: tfsdk.Plan{Schema: schemaResp.Schema, Raw: planVal},
+		}
+		r.(resource.ResourceWithModifyPlan).ModifyPlan(ctx, req, resp)
```

golangci-lint (2.12.2)

[error] type assertion must be checked

(forcetypeassert)